### PR TITLE
Added default values & color/B&W print option

### PR DESCRIPTION
**Pull request by pums12**
Set default values for PrintNodePrintJob and PrintNodePrintJobOptions

Added Print method to PrintNodeJob, so that you can print without having the query for the printers first if you already have the PrinterId.

Added PrintNodePrintJob constructor

### CODE REVIEW

File: PrintNodePrintJobOptions.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ public class PrintNodePrintJobOptions
         /// See <a href="https://www.printnode.com/docs/api/curl/#parameters">Parameters</a>.
         /// </summary>
         [JsonProperty("pages")]
-        public string Pages { get; set; }
+        public string Pages { get; set; } = "-";
```
Comment (christianz) [inline PrintNodePrintJobOptions.cs:76]: Have you tested this? I haven't looked very well, but can't find anything about setting Pages to "-" in the official documentation?

File: PrintNodePrintJobOptions.cs [hunk 2]
```diff
@@ -64,7 +64,7 @@ public class PrintNodePrintJobOptions
         /// See <a href="https://www.printnode.com/docs/api/curl/#printer-capabilities">Printer capabilities</a>.
         /// </summary>
         [JsonProperty("nup")]
-        public int Nup { get; set; }
+        public int Nup { get; set; } = 1;
```
Comment (christianz) [inline PrintNodePrintJobOptions.cs:67]: According to the official docs this should be an array of integers, so we might have another issue here. What's the reasoning/result of defaulting this to 1?

File: PrintNodePrintJob.cs [hunk 3]
```diff
@@ -136,6 +147,18 @@ public sealed class PrintNodePrintJob
             var list = JsonConvert.DeserializeObject<IEnumerable<IEnumerable<PrintNodePrintJobState>>>(response);
 
             return list.FirstOrDefault();
+        }
+
+		 public async Task<long> Print()
```
Comment (christianz) [inline PrintNodePrintJob.cs:152]: Is this necessary when you can use PrintNodePrinter.AddPrintJob()?

**Comment by christianz** [comment]
Hi, and thank you so much for the PR!

I don't have much time left over to maintain this repo these days, but it's awesome to see that people are using it and are still adding to it. I've added some comments asking about your reasoning on some of the changes.

**Comment by pums12** [comment]
Yes, I have tested it but only with a single page.

It is documented in Print Job options under Pages  "- prints all pages."
https://www.printnode.com/docs/api/curl#printjob-creating

**Comment by pums12** [comment]
The print capabilities return an array for Nup but for the print, it asks
for an Integer. The default they document is 1.

**Comment by pums12** [comment]
I wanted to be able to print a job without having to query PrintNode for
the printer first.

I already had the ID from Print Dialogue I had shown the user.

**Comment by pums12** [comment]
Hi,

No problem.

Thanks for taking the time to build the library in the first place!

BTW: The Color option I have added for PrintJobs is undocumented on
PrintNode but they mentioned I could try in a support email and so far it
seems to work well (only on Print Engine 6 though).

Thanks,

Ian

File: PrintNodePrintJobOptions.cs (hunk 1 above)
Comment (pums12) [reply PrintNodePrintJobOptions.cs:76]: Yes, I have tested it but only with a single page.

It is documented in Print Job options under Pages  "- prints all pages."
https://www.printnode.com/docs/api/curl#printjob-creating

File: PrintNodePrintJobOptions.cs (hunk 2 above)
Comment (pums12) [reply PrintNodePrintJobOptions.cs:67]: The print capabilities return an array for Nup but for the print, it asks
for an Integer. The default they document is 1.

File: PrintNodePrintJob.cs (hunk 3 above)
Comment (pums12) [reply PrintNodePrintJob.cs:152]: I wanted to be able to print a job without having to query PrintNode for
the printer first.

I already had the ID from the Print Dialogue I had shown the user.

**Pull request merged**